src/mon/OSDMonitor.cc

```diff
@@ -4055,6 +4211,7 @@ int OSDMonitor::get_version(version_t ver, uint64_t features, bufferlist& bl)
     reencode_incremental_map(bl, features);
   }
   inc_osd_cache.add({ver, significant_features}, bl);
+  inc_cache->_update_used_bytes(bl.length());
```

Comment:
how does this account for buffers that are dropped? simple_lru::add() both adds the new buffer, and may trim old ones from the cache

Reply:
I have introduced a mechanism to detect when buffers are about to be dropped and then recalculate the used bytes for each cache. Please verify.